components/nexus-repository/src/main/java/org/sonatype/nexus/repository/proxy/ProxyFacetSupport.java

```diff
@@ -227,7 +228,8 @@ protected Content fetch(String url, Context context, Content stale) throws IOExc
       return result;
     }
     if (status.getStatusCode() == HttpStatus.SC_NOT_MODIFIED) {
-      indicateVerified(context, cacheInfo);
+      checkState(stale != null); // to receive 304 we must issue condtl GET (bad server?)
```

Comment:
is it worth moving this comment into the `checkState` call as a message to help with problem determination?